src/hotspot/share/opto/vectornode.hpp

```diff
@@ -778,6 +778,63 @@ class StoreVectorNode : public StoreNode {
                                                      idx == MemNode::ValueIn + 1; }
 };
 
+class StoreVectorMaskedNode : public StoreVectorNode {
+ public:
+  StoreVectorMaskedNode(Node* c, Node* mem, Node* dst, Node* src, const TypePtr* at, Node* mask)
+   : StoreVectorNode(c, mem, dst, at, src) {
+    assert(mask->bottom_type()->is_long(), "sanity");
+    init_class_id(Class_StoreVector);
+    set_mismatched_access();
+    add_req(mask);
+  }
+
+  virtual int Opcode() const;
+
+  virtual uint match_edge(uint idx) const {
+    return idx > 1;
+  }
+  Node* Ideal(PhaseGVN* phase, bool can_reshape);
+
+  static StoreVectorMaskedNode* make(int opc, Node* ctl, Node* mem, Node* dst, Node* src,
+                                    const TypePtr* atype, Node* mask);
+};
+
+class LoadVectorMaskedNode : public LoadVectorNode {
+ public:
+  LoadVectorMaskedNode(Node* c, Node* mem, Node* src, const TypePtr* at, const TypeVect* vt, Node* mask)
+   : LoadVectorNode(c, mem, src, at, vt) {
+    assert(mask->bottom_type()->is_long(), "sanity");
+    init_class_id(Class_LoadVector);
+    set_mismatched_access();
+    add_req(mask);
+  }
+
+  virtual int Opcode() const;
+
+  virtual uint match_edge(uint idx) const {
+    return idx > 1;
+  }
+  Node* Ideal(PhaseGVN* phase, bool can_reshape);
+
+  static LoadVectorMaskedNode* make(int opc, Node* ctl, Node* mem, Node* src,
+                                const TypePtr* atype, const TypeVect* vt,
+                                Node* mask);
+};
+
+class VectorMaskGenNode : public TypeNode {
+ public:
+  VectorMaskGenNode(Node* src, const Type* ty, const Type* ety): TypeNode(ty, 2), _elemType(ety) {
```

Comment:
Sorry, I don't quite understand the arguments here. What does 'src' mean to the mask?

Reply:
ty   -> Node type , long in this case since for X86 mask register is 64 bit wide.
ety -> Mask element type, currently used during LoadVectorMasked/StoreVectorMasked idealization to compute the block sizes for constant masks and replace masked vector operations with non-masked if block size is equal to vector size.

Reply:
Please, use meaningful names.